src/components/DeleteConfirmation.js

```diff
@@ -0,0 +1,23 @@
+import Popup from "./Popup.js";
+
+export default class DeleteConfirmation extends Popup {
+  constructor(popupSelector, deleteCardForm, handleDeleteCard) {
+    super({ popupSelector });
+    this._handleDeleteCard = handleDeleteCard;
+    this._deleteCardForm = deleteCardForm;
+  }
+
+  open(card) {
+    this._card = card;
+    super.open();
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._deleteCardForm.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this._handleDeleteCard(this._card);
+      super.close();
```

Comment:
You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well

Reply:
You haven't corrected my remark https://github.com/Parnika006/se_project_aroundtheus/pull/1#discussion_r1693271579